src/Tooltip.js

```diff
@@ -79,10 +89,24 @@ class Tooltip extends React.Component {
     };
   }
 
+  show() {
+    if (!this.props.isOpen) {
+      this.toggle();
+    }
+  }
+  hide() {
+    if (this.props.isOpen) {
+      this.toggle();
+    }
+  }
+
   handleDocumentClick(e) {
     if (e.target === this._target || this._target.contains(e.target)) {
-      if (this._hoverTimeout) {
-        clearTimeout(this._hoverTimeout);
+      if (this._hideTimeout) {
+        clearTimeout(this._hideTimeout);
+      }
+      if (this._showTimeout) {
+        clearTimeout(this._showTimeout);
```

Comment:
If fact, everywhere you use `clearTimeout` you will have this problem. It needs to clear the timeout and unset the var that contains the timeout id (`this._showTimeout` and `this.hideTimeout`).
Also, when the timeout triggers the function (`show`/`hide`) they need to clear the var containing their respective timeout id (`this._showTimeout` and `this.hideTimeout` respectfully)